openretina/callbacks/weight_visualization_callback.py

```diff
@@ -0,0 +1,96 @@
+import inspect
+import os
+from contextlib import contextmanager
+
+import torch
+from lightning.pytorch.callbacks import Callback
+
+
+@contextmanager
+def eval_state(model: torch.nn.Module):
+    training_status = model.training
+
+    try:
+        model.eval()
+        yield model
+    finally:
+        model.train(training_status)
+
+
+class ConditionalWeightVisualizationCallback(Callback):
+    """
+    PyTorch Lightning callback that calls the model's save_weight_visualizations
+    function conditionally based on epoch settings.
+    """
+
+    def __init__(
+        self,
+        every_n_epochs=1,
+        start_epoch=0,
+        call_on_train_epoch_end=True,
+        call_on_validation_epoch_end=False,
+        save_plots=False,
+        folder_path=None,
+    ):
```

Comment:
```suggestion
    def __init__(
        self,
        every_n_epochs: int = 1,
        start_epoch: int = 0,
        call_on_train_epoch_end: bool = True,
        call_on_validation_epoch_end: bool = False,
        save_plots: bool = False,
        folder_path: str | None = None,
    ):
```
If we put the type information from the docstring as type annotations these will be automatically checked for consistency.